src/Data/List/Lazy.purs

```diff
@@ -705,3 +706,11 @@ foldM f a xs =
          Nothing -> pure a
          Just { head: b, tail: bs } ->
                        f a b >>= \a' -> foldM f a' bs
+
+-- | Perform a right fold lazily
+foldr' :: forall a b. Z.Lazy b => (a -> b -> b) -> b -> List a -> b
```

Comment:
Well, something like this will compile:

```purescript
foldr' op = foldr op'                                                                                                                                              
  where op' a b = Z.defer \_ -> a `op` b
```

But using `foldr` kills the laziness. So the test, for example, will just freeze on its infinite input.

Thinking if there is any kind of trick to be done here.